src/drivers/amd/rembrandt/acp_bt_dma.c

```diff
@@ -0,0 +1,439 @@
+	} else {
+		tr_err(&acp_bt_dma_tr, "ACP:direction not defined %d", channel->direction);
+		return -EINVAL;
+	}
+	return 0;
+}
+
+/* fill in "status" with current DMA channel state and position */
+static int acp_dai_bt_dma_status(struct dma_chan_data *channel,
+		struct dma_chan_status *status,
+		uint8_t direction)
+{
+	/* nothing to do on renoir */
+	return 0;
+}
+
+/* set the DMA channel configuration
+ * source/target address
+ * DMA transfer sizes
+ */
+static int acp_dai_bt_dma_set_config(struct dma_chan_data *channel,
+				struct dma_sg_config *config)
+{
+	uint32_t tx_ringbuff_addr;
+	uint32_t rx_ringbuff_addr;
+	uint32_t fifo_addr;
+	acp_bt_tx_ringbufaddr_t         tx_rbuff;
+	acp_bt_tx_fifoaddr_t            tx_fifo;
+	acp_bt_tx_fifosize_t            tx_fifo_size;
+	acp_bt_tx_dmasize_t             tx_dma_size;
+	acp_bt_tx_ringbufsize_t         tx_rbuff_size;
+	acp_bt_tx_ringbufaddr_t         rx_rbuff;
+	acp_bt_tx_fifoaddr_t            rx_fifo_addr;
+	acp_bt_tx_fifosize_t            rx_fifo_size;
+	acp_bt_tx_dmasize_t             rx_dma_size;
+	acp_bt_tx_ringbufsize_t         rx_rbuff_size;
+	acp_bt_tx_intr_watermark_size_t tx_wtrmrk;
+	acp_bt_rx_intr_watermark_size_t rx_wtrmrk;
+
+	channel->is_scheduling_source = true;
+	channel->direction = config->direction;
+	switch (config->direction) {
+	case DMA_DIR_MEM_TO_DEV:
+		config->elem_array.elems[0].src =
+			(config->elem_array.elems[0].src & ACP_DRAM_ADDRESS_MASK);
+		tx_ringbuff_addr = (config->elem_array.elems[0].src | 0x01000000);
+		/* BT Transmit FIFO Address and FIFO Size */
+		fifo_addr = BT_TX_FIFO_ADDR;
+		tx_fifo.u32all =  fifo_addr;
+		io_reg_write((PU_REGISTER_BASE + ACP_P1_BT_TX_FIFOADDR), tx_fifo.u32all);
+		tx_fifo_size.u32all = ACP_BT_FIFO_BUFFER_SIZE;
+		io_reg_write((PU_REGISTER_BASE + ACP_P1_BT_TX_FIFOSIZE), tx_fifo_size.u32all);
+		/* Transmit RINGBUFFER Address and size */
+		tx_rbuff.u32all = tx_ringbuff_addr;
+		io_reg_write((PU_REGISTER_BASE + ACP_P1_BT_TX_RINGBUFADDR), tx_rbuff.u32all);
+		tx_rbuff_size.u32all = ACP_BT_FIFO_BUFFER_SIZE;
+		io_reg_write((PU_REGISTER_BASE + ACP_P1_BT_TX_RINGBUFSIZE),
+				tx_rbuff_size.u32all);
+		/* Transmit DMA transfer size in bytes */
+		tx_dma_size.u32all = ACP_DMA_TRANS_SIZE;
+		io_reg_write((PU_REGISTER_BASE + ACP_P1_BT_TX_DMA_SIZE), tx_dma_size.u32all);
+		/* Watermark size for BT transfer fifo */
+		tx_wtrmrk.bits.bt_tx_intr_watermark_size = ACP_BT_FIFO_BUFFER_SIZE/2;
+		io_reg_write((PU_REGISTER_BASE + ACP_P1_BT_TX_INTR_WATERMARK_SIZE),
+				tx_wtrmrk.u32all);
+		break;
+	case DMA_DIR_DEV_TO_MEM:
+		config->elem_array.elems[0].dest =
+			(config->elem_array.elems[0].dest & ACP_DRAM_ADDRESS_MASK);
+		rx_ringbuff_addr = (config->elem_array.elems[0].dest | 0x01000000);
+		fifo_addr = BT_RX_FIFO_ADDR;
+		/* BT Receive FIFO Address and FIFO Size*/
+		rx_fifo_addr.u32all =  fifo_addr;
+		io_reg_write((PU_REGISTER_BASE + ACP_P1_BT_RX_FIFOADDR),
+				rx_fifo_addr.u32all);
```

Comment:
why not do it as then you don't need to define some variables
````c
io_reg_write((PU_REGISTER_BASE + ACP_P1_BT_RX_FIFOADDR),
				fifo_addr);
````
And also some following usages since you don't change some bits of them